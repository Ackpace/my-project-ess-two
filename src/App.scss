@import './styles/mixins';

p,
span {
  color: var(--border-color);
} 
 
section+section {
  margin-top: 24px;
}