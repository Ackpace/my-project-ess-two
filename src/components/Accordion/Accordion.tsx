import { useEffect, useState } from 'react'
import classNames from 'classnames'
import './accordion.scss'

interface AccordionProp {
    children: JSX.Element
    title: string
    open?: boolean
}

const Accordion = ({ children, title, open }: AccordionProp) => {
    const [isActive, setIsActive] = useState(false)
    const contentClasses = classNames(
        { 'activated mt-4': isActive || open },
        { 'deactivated': !isActive }
    )

    useEffect(() => {
        if (open) setIsActive(true)
    }, [open])

    return (
        <div className='accordion-container'>
            <div className="accordion-title" onClick={() => setIsActive(!isActive)}>
                <h2 className='mb-1'>{title}</h2>
                <div>{isActive ? <span>&#9650;</span> : <span>&#9660;</span>}</div>
            </div>
            <div className={contentClasses}>{children}</div>
        </div>
    )
}


interface ContentProp {
    children: JSX.Element
}

const Content = ({ children }: ContentProp) =>
    <div>{children}</div>


Accordion.Content = Content

export default Accordion

Accordion.defaultProps = {
    open: false
}