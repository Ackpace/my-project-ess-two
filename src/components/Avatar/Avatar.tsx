
interface AvatarProp {
    img: string,
    idx: number
}
 
function Avatar(props: AvatarProp) {
    return (
        <div className={`avatar-img-container avatar-${props.idx}`}>
            <img src={props.img} alt='avatar' />
        </div>
    )
}

export default Avatar