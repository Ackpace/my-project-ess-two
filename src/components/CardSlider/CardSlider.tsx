import { useRef, useState, useEffect } from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faAngleLeft } from "@fortawesome/free-solid-svg-icons/faAngleLeft";
import { faAngleRight } from "@fortawesome/free-solid-svg-icons/faAngleRight";
import Loader from "../../components/Loader/Loader";
import Button, { ButtonTypes } from "../Button/Button";
import "./card-slider.scss";
 
interface CardSliderProps {
  children: JSX.Element[] | JSX.Element | undefined;
  isLoading: boolean;
}

const getWindowSize = () => {
  const { innerWidth } = window;
  return innerWidth;
};

const CardSlider = ({ children, isLoading }: CardSliderProps) => {
  const [windowSize, setWindowSize] = useState(getWindowSize);
  const [offset, setOffset] = useState(0);
  const [scrollX, setScrollX] = useState(0);
  const [isScrollEnd, setIsScrollEnd] = useState(false);
  const contentsRef = useRef<HTMLDivElement>(null);

  // called initially
  useEffect(() => {
    setScrollX(contentsRef.current?.scrollLeft ?? -1);
    // Update the state with new window width
    const handleResize = () => {
      setWindowSize(getWindowSize);
    };

    // Add a listener
    window.addEventListener("resize", handleResize);

    // Clean up the listener
    return () => {
      window.removeEventListener("resize", handleResize);
    };
  }, []);

  useEffect(() => {
    const refObject = contentsRef.current;
    refObject!.scrollLeft = scrollX;
    setIsScrollEnd(refObject!.scrollWidth <= scrollX + offset + 66);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [scrollX]);

  const scroll = (scrollOffset: number) => {
    setOffset(scrollOffset);
    const refObject = contentsRef.current;

    if (refObject!.scrollWidth > scrollX + 66) {
      setScrollX(scrollX + scrollOffset ?? -1);
    }
  };

  return (
    <div className="card-slider-container">
      {
        // scrollX round down bc it could be sth like 0.2
        Math.floor(scrollX) > 0 ? (
          <div className="scroll-button left">
            <Button
              type={ButtonTypes.circle}
              onClick={() => scroll((windowSize / 2) * -1)}
              children={<FontAwesomeIcon icon={faAngleLeft} />}
            />
          </div>
        ) : null
      } 

      <div className="contents space-x-4" ref={contentsRef}>
        {isLoading ? <Loader /> : children}
      </div>

      {!isScrollEnd && (
        <div className="scroll-button right">
          <Button
            type={ButtonTypes.circle}
            onClick={() => scroll(windowSize / 2)}
            children={<FontAwesomeIcon icon={faAngleRight} />}
          />
        </div>

      )}
    </div>
  );
};

export default CardSlider;

CardSlider.defaultProps = {
  isLoading: false
}