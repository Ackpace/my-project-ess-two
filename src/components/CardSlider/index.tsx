import React from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faAngleLeft } from '@fortawesome/free-solid-svg-icons/faAngleLeft'
import { faAngleRight } from '@fortawesome/free-solid-svg-icons/faAngleRight'

interface CardSliderProps {
    title: string
}

const index = (props: CardSliderProps) => {
    return (
        <div className='flex'>
            <FontAwesomeIcon icon={faAngleLeft} />
            {props.title}
            <FontAwesomeIcon icon={faAngleRight} />
        </div>
    )
}

export default index