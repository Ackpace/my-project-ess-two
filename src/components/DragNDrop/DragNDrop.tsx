import { useState } from 'react'
import Button from '../Button/Button'
import { getStorage, ref, uploadBytesResumable, getDownloadURL } from "firebase/storage"
import { getAuth } from "firebase/auth"
import { doc, updateDoc, deleteField } from "firebase/firestore";
import { db } from '../../firebase.config'

function DragNDrop() {
    const [file, setFile] = useState<File>()
    const [dragActive, setDragActive] = useState(false)
    const auth = getAuth()

    const handleFileInput = (e: React.ChangeEvent<HTMLInputElement>) => {
        e.preventDefault()
        if (e.target.files && e.target.files[0]) {
            setFile(e.target.files[0])
        }
    }

    const handleDrag = (e: React.DragEvent<HTMLDivElement>) => {
        e.preventDefault()
        e.stopPropagation()

        if (e.type === 'dragenter' || e.type === 'dragover') {
            setDragActive(true)
        } else if (e.type === 'dragleave') {
            setDragActive(false)
        }
    }

    const handleDrop = (e: React.DragEvent<HTMLDivElement>) => {
        e.preventDefault()
        e.stopPropagation()
        setDragActive(false)

        if (e.dataTransfer.files && e.dataTransfer.files[0]) {
            console.log(typeof e.dataTransfer.files[0])
            setFile(e.dataTransfer.files[0])
        }
    }

    const handleSubmit = async () => {
        if (file) {
            // delete avatar
            if (auth.currentUser) {
                const userRef = doc(db, 'users', auth.currentUser.uid)
                await updateDoc(userRef, {
                    avatar: deleteField()
                })
            }

            return new Promise((res, rej) => {
                const storage = getStorage()
                const fileName = `${auth?.currentUser?.uid}`

                const storageRef = ref(storage, 'images/' + fileName)

                const uploadTask = uploadBytesResumable(storageRef, file)
                uploadTask.on('state_changed',
                    (snapshot) => {
                        const progress = (snapshot.bytesTransferred / snapshot.totalBytes) * 100;
                        console.log('Upload is ' + progress + '% done');
                        switch (snapshot.state) {
                            case 'paused':
                                console.log('Upload is paused');
                                break;
                            case 'running':
                                console.log('Upload is running');
                                break;
                        }
                    },
                    (error) => {
                        rej(error)
                    },
                    () => {
                        // Handle successful uploads on complete
                        // For instance, get the download URL: https://firebasestorage.googleapis.com/...
                        getDownloadURL(uploadTask.snapshot.ref).then((downloadURL) => {
                            res(downloadURL)
                            console.log('File available at', downloadURL);
                        });
                    }
                );
            })

        } else {
            console.log('no file selected!!');
        }
    }

    return (
        <div className={`drag-n-drop-container ${dragActive ? 'drag-active' : ''}`} onDragEnter={handleDrag}>
            <input type='file' id='input-file-upload' hidden onChange={handleFileInput} />
            <label htmlFor='input-file-upload'>OR Choose Your Own</label>
            <span id='file-chosen' className='file-title'>{file ? file.name : 'No file selected.'}</span>
            <Button onClick={handleSubmit}>Submit</Button>
            {dragActive && <div className='drag-file-element' onDragEnter={handleDrag} onDragLeave={handleDrag} onDragOver={handleDrag} onDrop={handleDrop} ></div>}
        </div>
    )
}

export default DragNDrop