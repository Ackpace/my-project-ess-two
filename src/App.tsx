import {
  BrowserRouter as Router,
  Routes,
  Route,
  Navigate
} from 'react-router-dom'
import Home from './pages/Home'
import About from './pages/About'
import Shows from './pages/Shows'
import Layout from './components/Layout'
import './App.scss'

function App() {
  return (
    <Router>
      <Layout>
        <Routes>
          <Route path='/home' element={<Home />} />
          <Route path='/about' element={<About />} />
          <Route path='/tv-shows' element={<Shows />} />
          <Route path='/' element={<Navigate replace to='/home' />} />
        </Routes>
      </Layout>
    </Router>
  );
}

export default App;
