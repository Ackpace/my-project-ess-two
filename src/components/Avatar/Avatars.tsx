import { useState } from "react"
import Avatar from './Avatar'
import { avatarSrc } from '../../assets/avatarSrc'
import './avatars.scss'

interface AvatarsProp {
    userName?: string | null
    sendAvatarIdx: (idx: number) => void
}

function Avatars(props: AvatarsProp) {
    const fullName = props.userName
    const firstName = fullName && fullName.split(' ')[0]
    const imgSrc = avatarSrc

    const getAvatarIdx = (idx: number) => {
        props.sendAvatarIdx(idx)
    }

    return (
        <div className='avatars'>
            {props.userName && <h1>Hi, {firstName}</h1>}
            <h2>Choose your avatar</h2>
            <div className='avatar-selection'>
                {
                    imgSrc.map((obj, idx) => (
                        <div
                            onMouseEnter={() => getAvatarIdx(idx)}
                            onMouseLeave={() => props.sendAvatarIdx(-1)}
                        >
                            <Avatar img={obj.img} key={idx} idx={idx} />
                        </div>
                    ))
                }
            </div>
        </div>
    )
}

export default Avatars