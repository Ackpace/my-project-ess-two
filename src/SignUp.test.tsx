
import { getByLabelText, fireEvent, getByPlaceholderText, getByRole, getByText, render, screen, waitFor, getByTestId } from '@testing-library/react';
import SignUp from './pages/SignUp';
import userEvent from '@testing-library/user-event';
import exp from 'constants';
import { BrowserRouter } from 'react-router-dom';
import axios from 'axios';
import ContentDetails from './components/ContentDetails/ContentDetails'
import showToastErrMessage from './components/SignIn/SignIn';
import form from './pages/SignUp'
import TVShow from "./pages/TVShow";
import {  Pagination } from '@mui/material';
import register from './pages/SignUp'
import App from './App';
import SignIn from './components/SignIn/SignIn';
import Avatars from './components/Avatar/Avatar';
import Avatar from './components/Avatar/Avatar';
import Home from './pages/Home';
import ContentIntro from './pages/ContentInfo';
import CardSlider from './components/CardSlider/CardSlider';
import TmdbInput from './components/FormElements/TmdbInput';
import Accordion from './components/Accordion/Accordion';
import DragNDrop from './components/DragNDrop/DragNDrop';
import About from './pages/About';
import SliderMenu from './components/SliderMenu/SliderMenu';


const stubbedCountry = {
    poster_path: "/73gIfV8gDwkVgUrFDzt4TfZC9Zc.jpg",
     title: "It Lives Inside",
     release_date: "2023-09-06",
      vote_average: "5.9",
    original_title: "It Lives Inside",
     overview: "Desperate to fit in at school, Sam rejects her",
    backdrop_path: "/dp0rdbGSbZTyWumnbWspXQp87hA.jpg",
};

it("matches snapshot", () => {
  const { asFragment } = render(<ContentDetails movie={stubbedCountry} />);

  expect(asFragment()).toMatchSnapshot();
});
 



test("Login Form Heading present", () => {
  render(
    <BrowserRouter><SignUp /> </BrowserRouter>).container;
  //get by texta
  const headingelement = screen.getByTestId("signupform");
  expect(headingelement).toBeInTheDocument();
});



test("Check if Text Box for username is present by placeholder text", () => {
  render(
    <BrowserRouter><SignUp /> </BrowserRouter>).container;
  //get by role with parameters
  //get by placeholder text
  const passwordInput = screen.getByPlaceholderText("Jay Oh");
  expect(passwordInput).toBeInTheDocument();
});


test("Check if Text Box for email is present by placeholder text", () => {
  render(
    <BrowserRouter><SignUp /> </BrowserRouter>).container;

  //get by role with parameters
  //get by placeholder text
  const passwordInput = screen.getByPlaceholderText("JayIsAwesome@gmail.com");
  expect(passwordInput).toBeInTheDocument();
});



/*
jest.mock('./pages/Home', () => ({
  fetchPosts: jest.fn().mockReturnValue(Promise.resolve([])),
}));

describe('The DataViewer component', () => {
  describe('when provided with a dictionary', () => {
    it('should render a ReactJson component', async () => {
      const dataViewer = render(<Home/>);
      await dataViewer.findByTestId('react-json-view');
    });
  });
  describe('when provided with a string', () => {
    it('should render a paragraph with the provided string', async () => {
      const text = 'value';
      const dataViewer = render(<Home />);
      await dataViewer.findByText(text, { selector: 'p' });
    });
  });
});
*/

jest.mock('./components/CardSlider/CardSlider')

test('ParentComponent rendered', () => {
  render(<BrowserRouter><ContentIntro /> </BrowserRouter>).container
  expect(screen.getByTestId("custom-element")).toBeInTheDocument()
})

test('ChildComponent mock was called', () => {
  render(<BrowserRouter><ContentIntro /> </BrowserRouter>).container
  expect(CardSlider).toBeTruthy()
})

/*
describe("Test the ContentDetails Component",  ()=>{
  test("render the ContentDetails informations as h elements", async () =>{
    render(
      <DragNDrop/> ).container;
      const hList = await screen.findByRole("div");
      expect(hList).toHaveValue(1);
  })
})*/


describe("About Component",  ()=>{
  test("should render About Component correctly", () =>{
    render(
      <BrowserRouter><About /> </BrowserRouter>).container;
      const hList = screen.getByTestId("aboutdiv");
      expect(hList).toBeInTheDocument();
  })
})



it('renders an error with invalid password', () => {
  
  render(<BrowserRouter><SignIn /> </BrowserRouter>).container;
  const passwordField = screen.getByPlaceholderText('ex) 123123')
  const submitButton = screen.getByText('Sign in')
  


  fireEvent.focus(passwordField)
  fireEvent.change(passwordField, { target: { value: '1234' } })
  fireEvent.blur(passwordField)
 
  fireEvent.click(submitButton)
  expect(passwordField).toHaveTextContent("")
  
 })

/*
 describe('Student Toggle', () => {
  it('should display student', () => {
    const eventHandler = jest.fn();
    render(<BrowserRouter><SignUp /> </BrowserRouter>).container;
    const { getByTestId } = render(<form onClick={eventHandler}/>);

    
     const button = getByTestId('registebutton');
     fireEvent.click(button);
    

    expect(eventHandler).toHaveBeenCalled();
    //how do I test? <div style={studentDisplay ? {display:'block'} : {display:'none'}}>
  });
});
*/

///nazov///
  test("should render About Component correctly", () =>{
    render(
      <BrowserRouter><DragNDrop /> </BrowserRouter>).container;
      const hList = screen.getByRole("divn");
      expect(hList).toBeInTheDocument();
      expect(hList).toHaveTextContent("Submit")
  })


/*
  test("should display a blank login form, with remember me checked by default", async () => {
    render(
      <BrowserRouter><SignUp /> </BrowserRouter>).container;
  
    const loginForm = screen.findByTestId("login-form");
  
    expect(loginForm).toHaveFormValues({
      name: "",
      email: "",
      password: ""
    });
  });*/









// describe('expectedData', () => {
//     it('checks if returned data from API rendered into component', async () => {
//         render(<Genres genres setGenres selectedGenres setSelectedGenres />);

//         await waitFor(() => {
//             expect(screen.getByText("/value from the api")).toBeInTheDocument();
//         });
//     });
// });















//nazov z videa//
describe("Test the ContentDetails Component",  ()=>{
        test("render the ContentDetails informations as h elements", () =>{
          
          render(
            <BrowserRouter><DragNDrop /> </BrowserRouter>).container;
            const hList = screen.getByRole("divn");
            expect(hList).toHaveAttribute("role","divn")
        })
    })


/*    
    describe("Test the ContentDetails Component",  ()=>{
      test("render the ContentDetails informations as h elements", () =>{
        
        render(
          <BrowserRouter><SliderMenu /> </BrowserRouter>).container;
          const divElement = screen.getByRole("contentinfo")
          const buttonElement = screen.getByTestId("slidermenubutton")
          fireEvent.click(buttonElement)

          expect(divElement).toHaveTextContent("")
      })
  })
*/

/*
  it('renders an error with invalid password', () => {
    render(<BrowserRouter><SignIn /> </BrowserRouter>).container;
    const passwordField = screen.getByPlaceholderText('ex) 123123')
    const submitButton = screen.getByText('Sign in')
   
    fireEvent.focus(passwordField)
    fireEvent.change(passwordField, { target: { value: '1234' } })
    fireEvent.blur(passwordField)
   
    fireEvent.click(submitButton)
    expect(showToastErrMessage).toHaveTextContent("Incorrect Username/Password")
   })
*/

/*
describe("Test the ContentDetails Component", ()=>{
        test("render the ContentDetails informations as h elements", async () =>{
            const { getByText } = render(<SignUp />);
             const helloWorldElement = getByText("Hello World");
            expect(helloWorldElement).toBeInTheDocument();
        })
    })

    test("increments count on button click", () => {
        const { getByTestId } = render(<Pagination count={9} />);
        const countElement = getByTestId("count");
      
        expect(countElement.textContent).toBe("9");
      });

      test("increments count on button click", () => {
       render(<Pagination count={9} />);
       
      
       expect(screen.queryByTestId('count')).not.toBeInTheDocument();
    });

    import { BrowserRouter as Router } from "react-router-dom";

    test("renders about component when visiting the about route", () => {
        render(
          <Router >
            <App />
          </Router>
        );
      
        expect(screen.getByText("This is the About component")).toBeInTheDocument();
      });

*/
















    
// test("password input should have type password",()=>{
//     render(<SignUp/>)
//     const password = screen.getByPlaceholderText("123123");
//     expect(password). getByPlaceholderText("Password");
// });


// test("Login Form Heading present", () => {
//     render(<SignUp />);
//     //get by texta
//     const headingelement = screen.getByText(/join us/i);
//     expect(headingelement).toBeInTheDocument();
//   });























