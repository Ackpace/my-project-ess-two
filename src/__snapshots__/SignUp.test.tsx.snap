// Jest Snapshot v1, https://goo.gl/fbAQLP
 
exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="movie_card"
    id="bright"
  >
    <div
      class="info_section"
    >
      <div
        class="movie_header"
      >
        <img
          class="locandina"
          src="https://image.tmdb.org/t/p/w500//73gIfV8gDwkVgUrFDzt4TfZC9Zc.jpg"
        />
        <h1>
          It Lives Inside
        </h1>
        <h4>
          2023-09-06
        </h4>
        <span
          class="minutes"
        >
          5.9
        </span>
        <p
          class="type"
        >
          It Lives Inside
        </p>
      </div>
      <div
        class="movie_desc"
      >
        <p
          class="text"
        >
          Desperate to fit in at school, Sam rejects her
        </p>
      </div>
      <div
        style="display: flex; margin: 5px; justify-content: flex-start;"
      >
        <div
          style="margin: 5px;"
        >
          <li>
            <button
              class="card_btn"
            >
              Watch Trailer
            </button>
          </li>
        </div>
      </div>
    </div>
    <div>
      <img
        class="blur_back"
        src="https://image.tmdb.org/t/p/w500//dp0rdbGSbZTyWumnbWspXQp87hA.jpg"
      />
    </div>
  </div>
</DocumentFragment>
`;
